### ISSUE TITLE
VG Giraffe indexing terminating with Warnings (complex insertions not canonicalizeable) [canonicalize]

### ISSUE THREAD

**Issue by kvaddad1** [vgteam/vg#3917]
Hi, 
I am trying to build VG index using giraffe pipeline on 1KGP_phase3 VCF files on selected samples. The indexing command is terminating after throwing a bunch of warnings as follows.
`Warning: insertion END and POS do not agree (complex insertions not canonicalizeable) [canonicalize]`

There was no error message as such, the command abruptly ends after showing these warnings and the indexing command was not successful. Can you please guide me on what could be the issue here.  I am using autoindex command as follows 
`vg autoindex --workflow giraffe -r ${ref} "${VCF_ARGS[@]}" -p parent --threads 32
`
and my vg version is `vg: variation graph tool, version v1.46.0 "Altamura"`
Please suggest. 
Thank you.

**Comment by jeizenga** [comment]
Are you able to share the data that you used when you ran into the error?

**Comment by kvaddad1** [comment]
Hi @jeizenga,
Thank you so much for your kind response. I am trying to run autoindex giraffe pipeline with GRCh38.fasta (GCA_no_alt_version) along with th chromosome vcf files repository available at 
`http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/20220422_3202_phased_SNV_INDEL_SV/`

Please suggest. Thank you so much.

**Comment by jeizenga** [comment]
And where did you get the HG38 version that you used as input?

**Comment by kvaddad1** [comment]
Hi @jeizenga 
The fasta I tried with the above VCF files is `GCA_000001405.15_GRCh38_no_alt_analysis_set.fna` and 
available at 
`https://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/001/405/GCA_000001405.15_GRCh38/seqs_for_alignment_pipelines.ucsc_ids/`

Please suggest, thank you so much for following up.

**Comment by adamnovak** [comment]
I think this might be a duplicate of #3922 and thus also fixed by #3933, according to the log @jeizenga gave me, which ended with:

```
Inversion can start after 53964
Inversion can end by inverting 54062
Buffered final node becomes: 34933728
vg: src/io/load_proto_to_graph.cpp:239: vg::io::load_proto_to_graph(vg::MutablePathMutableHandleGraph*, const std::function<void(const std::function<void(vg::Graph&)>&)>&)::<lambda(vg::Graph&)>: Assertion `get_handle(m.position().node_id(), m.position().is_reverse(), visited)' failed.
ERROR: Signal 6 occurred. VG has crashed. Visit https://github.com/vgteam/vg/issues/new/choose to report a bug.
Stack trace path: /tmp/vg_crash_s7GiJY/stacktrace.txt
Please include the stack trace file in your bug report!
```

But he said he also had to do something with vcflib to get it to accept the vcfs, so I am checking myself with the vcflib changes on top of that PR's to confirm that this now works.

**Comment by adamnovak** [comment]
My fix for #3922 also seems to be effective here, so I am going to close this as fixed already. I really need #3939 to do this run in a reasonable amount of time, though, so that will also be useful here.
